src/Featurizers/Components/GlobalMinimumFrequencyEstimator.h

```diff
@@ -0,0 +1,222 @@
+    using FrequencyType                     = std::chrono::system_clock::duration;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    FrequencyType                           Value;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    GlobalMinimumFrequencyAnnotation(FrequencyType value);
+    ~GlobalMinimumFrequencyAnnotation(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(GlobalMinimumFrequencyAnnotation);
+};
+
+namespace Details {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         GlobalMinimumFrequencyTrainingOnlyPolicy
+///  \brief         `GlobalMinimumFrequencyEstimator` implementation details.
+///
+template <typename EstimatorT>
+class GlobalMinimumFrequencyTrainingOnlyPolicy {
+public:
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using InputType                     = std::tuple<std::string, std::chrono::system_clock::time_point>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    static constexpr char const * const     NameValue = GlobalMinimumFrequencyEstimatorName;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    GlobalMinimumFrequencyTrainingOnlyPolicy(void);
+
+    void fit(InputType const &input);
+    GlobalMinimumFrequencyAnnotation complete_training(void);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    using FrequencyType                     = std::chrono::system_clock::duration;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    GrainedFrequencyEstimator                      _estimator;
+};
+}
+/////////////////////////////////////////////////////////////////////////
+///  \typedef       GlobalMinimumFrequencyEstimator
+///  \brief         A training-only class that finds the min, max
+///
+template <
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class GlobalMinimumFrequencyEstimator :

```

Comment:
>GlobalMinimumFrequencyEstimator [](start = 6, length = 31)

I didn't expect to see this implemented in this way - I thought that it would rely on annotations produced by the GrainedFrequencyEstimator rather that contain it. We have discussed how the LagLead/RollingWindow featurizer would work in relationship to the TimeSeriesImputer, but didn't expect to see those concepts extend to these components.